pkg/sql/opt/norm/rules/scalar.opt

```diff
@@ -441,3 +441,12 @@ $value
 (Indirection $input:* $index:* & (IsJSON $input))
 =>
 (FetchVal $input $index)
+
+
+[NormalizeLikePattern, Normalize]
```

Comment:
I think this rule (and the corresponding tests and golang functions) belongs in comp.opt instead.